src/Build/Instance/TaskRegistry.cs

```diff
@@ -1425,6 +1431,10 @@ private bool GetTaskFactory(TargetLoggingContext targetLoggingContext, ElementLo
                                     return false;
                                 }
                             }
+                            catch(InvalidProjectFileException)
+                            {
+                                throw;
```

Comment:
Comment, please :)

I'm also not sure I think it's worth the unique error code for this if the message is good. But it's already done so go with it.